### ISSUE TITLE
Transforms with random size incompatible with default collater of data loader

### ISSUE THREAD

**Issue by rijobro** [Project-MONAI/MONAI#1576]
**Describe the bug**
Transforms that produce images of variable size are incompatible with default collator, since this expects all images to be the same size.

Examples that I've found so far are (I guess same would apply to array versions)):
- `RandSpatialCropd` (with `random_size==True`)
- `RandRotated` (with `keep_size==False`)
- `RandZoomd` (with `keep_size==False`)
- `RandRotate90d` (if input wasn't square/cubic)

You wouldn't see this problem if you a) specify some custom collator, or b) followed up these transforms with other transforms that unified image shape (e.g., `SpatialPadd`). I can't imagine many people do the former, and for the latter I don't see the utility of allowing the image size to change if you are then going to crop/pad back to a fixed size. 

Unless I've mistaken something, I suggest removing functionality that could result in images inside the same batch having different sizes.

**To Reproduce**
```python
import numpy as np
from monai.data import CacheDataset, DataLoader
from monai.transforms import RandSpatialCropd, RandRotated

im = np.arange(0, 10 * 9).reshape(1, 10, 9)
data = [{"image": im} for _ in range(2)]

# alternate commenting these two lines
transform = RandSpatialCropd("image", roi_size=[8, 7], random_size=True)
transform = RandRotated("image", prob=1, range_x=np.pi, keep_size=False)

dataset = CacheDataset(data, transform)
loader = DataLoader(dataset, batch_size=2)
for _ in loader:
    pass
```

Example error:
```
stack expects each tensor to be equal size, but got [1, 13, 13] at entry 0 and [1, 10, 11] at entry 1
  File "/Users/rich/Documents/Code/MONAI/monai/data/utils.py", line 245, in list_data_collate
    return default_collate(data)
  File "/Users/rich/Desktop/text.py", line 14, in <module>
    for _ in loader:
```

**Comment by Nic-Ma** [comment]
Hi @rijobro ，

Thanks for your suggestion.
In the MONAI DataLoader, we have a customized `collate_fn` callback, maybe we can enhance it for your case?
https://github.com/Project-MONAI/MONAI/blob/master/monai/data/utils.py#L231

Thanks.

**Comment by rijobro** [comment]
Yes, when doing batch inverse transforms, the output will be different sizes if the original images were of different sizes. This can be solved by disabling collation (`lambda x: x`). To help users, I tried to add a useful error message [here](https://github.com/Project-MONAI/MONAI/pull/1514/files#diff-3e89dd0dfa0e28a491abed95ee5fd6321dbb0f1479ed820f7d9291247c4e6bb5R56-R60).

```python
try:
    return next(iter(inv_loader))
except RuntimeError as re:
    re = str(re)
    if "stack expects each tensor to be equal size" in re:
        re += "\nMONAI hint: try creating `BatchInverseTransform` with `collate_fn=lambda x: x`."
    raise RuntimeError(re)
```

I suppose we could do something similar here, where if an error occurs then we can suggest they use an alternative collator. And I would suggest implementing an alternative collator that pads all images to the same size perhaps? Using `DivisblePad` maybe. That, or no collation at all (`lambda x: x`). @ericspod ?

**Comment by wyli** [comment]
addressed by #1626
